light-client/src/components/scheduler.rs

```diff
@@ -21,51 +32,81 @@ where
     fn schedule(
         &self,
         light_store: &dyn LightStore,
-        next_height: Height,
+        current_height: Height,
         target_height: Height,
     ) -> Height {
-        self(light_store, next_height, target_height)
+        self(light_store, current_height, target_height)
     }
 }
 
+/// Basic scheduler which picks the appropriate midpoint without
+/// optimizing for performance using the blocks available in the light store.
+///
+/// ## Precondition
+/// - The light store contains at least one verified block. [LCV-SCHEDULE-PRE.1]
+///
+/// ## Postcondition
+/// - The resulting height must be valid according to `valid_schedule`. [LCV-SCHEDULE-POST.1]
 #[pre(light_store.latest(VerifiedStatus::Verified).is_some())]
-#[post(valid_schedule(ret, target_height, next_height, light_store))]
-pub fn schedule(
+#[post(valid_schedule(ret, target_height, current_height, light_store))]
+pub fn basic_schedule(
```

Comment:
Since this is actually performing bisection does it make sense to call it `bisecting_schedule`?

Reply:
Sure, that's indeed a better name. I chose `basic` here to express that this scheduler implementation does not do anything fancy for performance (like picking heights already present in the store, etc.)

Reply:
So perhaps, `basic_bisecting_schedule` or something like that?